### ISSUE TITLE
Vagrant triggers for resume never fire if called with vagrant up

### ISSUE THREAD

**Issue by noaho** [hashicorp/vagrant#11449]
### Vagrant version
Vagrant 2.2.7

### Host operating system
MacOS 10.15.3

### Guest operating system
bento/Ubuntu-18.04

Hi, I am having the same problem listed as closed in - [GH-6615]

I am trying to trigger an action based on machine resume, but the action only triggers if I explicitly resume the machine eg vagrant resume, it doesn't trigger when I vagrant up, even though I see the text "Resuming suspended VM..."

I have tried three ways, as a trigger, command, and a hook;
``` 
  config.vm.box = "bento/ubuntu-18.04"

  config.trigger.before :resume do |t|
    t.warn = "Before :resume"
  end

  config.trigger.before :resume, type: :command do |t|
    t.warn = "Before :resume command"
  end

  config.trigger.before :machine_action_resume, type: :hook do |t|
    t.warn = "Before :machine_action_resume hook!"
  end
```

### Actual behavior
When I run vagrant resume I get this output:
```
==> vagrant: You have requested to enabled the experimental flag with the following features:
==> vagrant:
==> vagrant: Features:  typed_triggers
==> vagrant:
==> vagrant: Please use with caution, as some of the features may not be fully
==> vagrant: functional yet.
==> vagrant: Running command triggers before resume ...
==> vagrant: Running trigger...
==> vagrant: Before :resume command
==> default: Running action triggers before resume ...
==> default: Running trigger...
==> default: Before :resume
==> vagrant: Running hook triggers before machine_action_resume ...
==> vagrant: Running trigger...
==> vagrant: Before :machine_action_resume hook!
==> vagrant: Running hook triggers before machine_action_resume ...
==> vagrant: Running trigger...
==> vagrant: Before :machine_action_resume hook!
==> vagrant: Running hook triggers before machine_action_resume ...
==> vagrant: Running trigger...
==> vagrant: Before :machine_action_resume hook!
==> default: Resuming suspended VM...`
```

When I run vagrant up (after vagrant suspend) I get this output:
```
==> vagrant: You have requested to enabled the experimental flag with the following features:
==> vagrant:
==> vagrant: Features:  typed_triggers
==> vagrant:
==> vagrant: Please use with caution, as some of the features may not be fully
==> vagrant: functional yet.
Bringing machine 'default' up with 'parallels' provider...
==> default: Resuming suspended VM...
```

### Expected behavior
Result should be the same regardless, if the VM is resumed. 

I've also tried this, but it never triggers. 

```
config.trigger.before :"Vagrant::Action::VM::Resume", type: :action do |t|
  t.warn = "Before vagrant Action"
end
```

### Steps to reproduce

1. vagrant up
2. vagrant suspend
3. vagrant up

### Vagrantfile
```ruby
Vagrant.configure("2") do |config|
  config.vm.box = "bento/ubuntu-18.04"

  config.trigger.before :resume do |t|
    t.warn = "Before :resume"
  end

  config.trigger.before :resume, type: :command do |t|
    t.warn = "Before :resume command"
  end

  config.trigger.before :machine_action_resume, type: :hook do |t|
    t.warn = "Before :machine_action_resume hook!"
  end

  # Doesn't trigger
  config.trigger.before :"Vagrant::Action::VM::Resume", type: :action do |t|
    t.warn = "Before vagrant Action"
  end
end
```

### Debug output
https://gist.github.com/noaho/76c96fbca7db531729e59fb96871baaf

### References
- GH-6615

**Comment by briancain** [comment]
Hey there @noaho - It looks like the specific `:action` and `:hook` type trigger has a bug based on its current implementation for certain actions and hooks...but for you once it's actually working, will want the action `:"VagrantPlugins::Parallels::Action::Resume"` instead of what you have there. Also, I don't believe there is a `:machine_action_resume` defined, Vagrant just calls `:machine_action_up`...however that action is called multiple times within a single `vagrant resume` or `vagrant up`, so I imagine that's also not exactly what you want.

That being said, the trigger command `:resume` _should_ work, but it will run before or after the entire Vagrant command `vagrant suspend`. So depending on what you're trying to do, that might not work for you.

Anyway, thanks again for trying out these experimental features! Sorry they aren't fully working at the moment :pray: We should have a related fix coming down the road soon that will solve these issues with hook and action triggers. :+1:

**Comment by noaho** [comment]
Thanks. 
I want to target this cross-platform however, not Parallels specific. I'm looking for a function that will run before the text "==> default: Resuming suspended VM...`"

I'm not sure what you mean when you say :resume will run after vagrant suspend? Shouldn't it run after vagrant resume and vagrant up? or am I missing something?

The objective here is to call vagrant reload when resumed from suspend, as there's some apps that really don't like being suspended... And I find myself in an unknown state quite often because of this.

**Comment by briancain** [comment]
> Thanks.
> I want to target this cross-platform however, not Parallels specific. I'm looking for a function that will run before the text "==> default: Resuming suspended VM...`"
> 
> I'm not sure what you mean when you say :resume will run after vagrant suspend? Shouldn't it run after vagrant resume and vagrant up? or am I missing something?

Sorry, I meant to say it would run before or after the command `vagrant resume` :+1: 

> 
> The objective here is to call vagrant reload when resumed from suspend, as there's some apps that really don't like being suspended... And I find myself in an unknown state quite often because of this.

What are you trying to run exactly when reloading? Perhaps you don't need something as specific as a hook or action for your needs here.

**Comment by noaho** [comment]
> What are you trying to run exactly when reloading? Perhaps you don't need something as specific as a hook or action for your needs here.

My servers in the wild don't go in to suspend - my vagrant vms do. 

Thus at the very least I would like to output a warning message so it becomes clear that the VM has been brought up from standby, not up from a clean boot as otherwise assumed.

**Comment by briancain** [comment]
Hey @noaho - Makes sense. @chrisroberts is working on some enhancements that will hopefully improve and fixup the action triggers. Hopefully this will give you some better control over what you're trying to do. It should go out in the next release.

**Comment by briancain** [comment]
Hey there! The enhancements patch that I had mentioned for vagrant triggers was merged a bit ago (#11455), and should be making its way into the next Vagrant release (2.2.8). I'm going to go ahead and close this as it should resolve the issue. If you still are experiencing an issue using that version feel free to re-open this one, or open a new one! Thanks.
